### ISSUE TITLE
Avalonia.FuncUI v0.5

### ISSUE THREAD

**Issue by sleepyfran** [fsprojects/Avalonia.FuncUI#189]
This is issue tracks the steps necessary for the release of the new version, which includes mainly the newest Avalonia package (#177) and the biggest change to FuncUI with the new React/Sutil state management via components (#179).

### Before release
- [ ] Change the package name (#173) 

### After release
- [ ] Update docs to contain the new package name and examples of both Elmish and Components (#174)
- [ ] Update README to provide examples with the new components
- [ ] Update templates to contain both Elmish and Components templates

**Comment by sleepyfran** [comment]
Folks, I was thinking to maybe skip the package renaming and publish 0.5 straight away and just release the new naming whenever it's done? It seems like it's going to take a bit, so just to not delay the addition of all the new goodies any further. What do you think @JaggerJo @JordanMarr?

**Comment by JaggerJo** [comment]
Yeah, good idea. Better get this out soon.

**Comment by JordanMarr** [comment]
I wholeheartedly agree!

**Comment by sleepyfran** [comment]
Awesome! I'll start the process tomorrow then whenever I find some time 😄

**Comment by sleepyfran** [comment]
Started the process in #195, please take a look 😃  With this merged the new version should be published automatically and also includes a fix for the current issue blocking the master CI and some README improvements + a counter app example in components.

@JordanMarr any chance you'd have some time to either update the docs or the templates to include components examples?

**Comment by JordanMarr** [comment]
I’ve been preoccupied this weekend working on a different project, but I can make some modifications to the docs over the next few days. 
I saw you made some changes already. Is there anything specifically that needs to be added?

**Comment by sleepyfran** [comment]
> I’ve been preoccupied this weekend working on a different project, but I can make some modifications to the docs over the next few days.
> I saw you made some changes already. Is there anything specifically that needs to be added?

Sure, no worries! I think we can release the 0.5 as it is (there's a huge amount of examples to take a look at) and update the docs and templates as we go, so there's no rush 😃 

I meant the docs that [are hosted here](https://avaloniacommunity.github.io/Avalonia.FuncUI.Docs/), they're inside of the `AvaloniaCommunity` repo and they don't contain any references to the new component system, so I think we should update them and once we have the templates done we should also add some tutorials in there similar to the ones that already there.

**Comment by AngelMunoz** [comment]
It would be nice to have these changes released, Avalonia was featured in the dotnet rocks podcast recently (https://dotnetrocks.com/?show=1785) and even FuncUI was mentioned, I can keep updating the old docs with the new changes (as well as a couple of blog posts) until we can get proper docs set up here (old docs are just markdown files anyway so migrating them shouldn't be difficult once we're there) but IMO I wouldn't wait for the docs to get these changes released

**Comment by sleepyfran** [comment]
> It would be nice to have these changes released, Avalonia was featured in the dotnet rocks podcast recently (https://dotnetrocks.com/?show=1785) and even FuncUI was mentioned (albeit briefly), I can keep updating the old docs with the new changes (as well as a couple of blog posts) until we can get proper docs set up here (old docs are just markdown files anyway so migrating them shouldn't be difficult once we're there) but IMO I wouldn't wait for the docs to get these changes released

Yeah, agree, I wasn't implying to wait until the docs changes were done, I was actually thinking to update them as we can after we release it 😃 

@JordanMarr @JaggerJo can I get a review on the PR so that we can have it released?

**Comment by JordanMarr** [comment]
Oh FFS, the whole thing was waiting on me. 
Sorry about that! Approved. 😅

**Comment by JordanMarr** [comment]
Why does a maintainer even have to be approved anyways?? It seems like a an unnecessary formality to me that just slows things down. 🤷‍♂️

**Comment by sleepyfran** [comment]
> Why does a maintainer even have to be approved anyways?? It seems like a an unnecessary formality to me that just slows things down. 🤷‍♂️

Agreed. We can always merge to master directly for quick stuff, but it's definitely better to have PRs for traceability. @JaggerJo do you by any change have permissions to change that in the repo? I barely have any options in the settings page of the repo.

**Comment by sleepyfran** [comment]
Also I just realized that the F# official page points to the wiki of this project instead of the actual docs:
https://fsharp.org/use/desktop-apps/

Do you think we should keep stuff in the documentation page that is in place right now, move everything to GitHub Wiki or support both? I think having two sources of truth will definitely make it harder to keep everything in sync, but also not sure which of the options is better.

**Comment by AngelMunoz** [comment]
I think the links to the wiki were made before the docs were a thing.
I wouldn't count on them we can make a PR there to update those links when we have the docs in place, I think I can still merge PR's on the old docs website so if needed I can still merge new content, but I think the ideal would be to set up docs here in the repository just like the other fsprojects

**Comment by JaggerJo** [comment]
Maybe moving everything to GitHub wiki is the best thing to so mid term?

Think it makes changing the Docs simpler.

**Comment by JaggerJo** [comment]
Would be nice if we could have a small FuncUI section in the official AvaloniaUI Docs.

Just to raise awareness that FuncUI exists.

**Comment by AngelMunoz** [comment]
> Maybe moving everything to GitHub wiki is the best thing to so mid term?
> 
> Think it makes changing the Docs simpler.


I don't see anything wrong with the current docs at least on the meantime, they are just markdown files and when you merge a change it gets deployed to github actions, you don't even need to download the repo for that

I think we could cut the wiki directly and start working on the fsdocs integration (if anyone has experience with that) and just migrate the entries directly to that format otherwise we can keep updating the existing docs with the new stuff until fsdocs is setup or simply just keep things as is

I will try to add the components entry over the weekend as wellas a blog post if I have a chance (together with the updated templates)

**Comment by AngelMunoz** [comment]
> Would be nice if we could have a small FuncUI section in the official AvaloniaUI Docs.
> 
> Just to raise awareness that FuncUI exists.

here's a somewhat crazy idea, what if we request the Avalonia Docs, to allow us to write FuncUI docs there as well?
IMO they are still avalonia docs right?


other than that, I guess also a simple entry wouldn't hurt

**Comment by JaggerJo** [comment]
We talked to the Avalonia Team and @danwalmsley said we could have a FuncUI section in the official docs!
